src/app/components/file-uploader/file-uploader.component.ts

```diff
@@ -0,0 +1,195 @@
+  @Input() acceptedTypes: Array<String>;
+
+  /**
+   * The maximum size of a file in bytes
+   */
+  private maxFileSize = 2097152;
+
+  /**
+   * The maximum size of a file in a readable format
+   */
+  private maxFileSizeReadable: string = this.formatBytes(this.maxFileSize, 0);
+
+  constructor(private uploadService: FileUploaderService,
+              private alertService: AlertService,
+              private fileRetrieverService: FileRetrieverService) { }
+
+  files: Array<UploadFile> = new Array<UploadFile>();
+
+  /**
+   * handle onFileDrop event
+   */
+  public onFileDropped(event): void {
+    this.prepareFilesList(event);
+  }
+
+  /**
+   * handle file from the file explorer
+   */
+  public fileBrowseHandler(files): void {
+     if (files.files !== this.files) {
+      this.prepareFilesList(files.files);
+     }
+  }
+
+  /**
+   * Delete file from files list
+   * @param index The index of the file to delete
+   */
+  public deleteFile(index: number): void {
+    this.files.splice(index, 1);
+    this.fileInput.nativeElement.value = '';
+  }
+
+
+  /**
+   * This will finalize the data from the uploadForm and do some filetype and -size checks.
+   * @param files The uploaded files from the uploadForm
+   */
+  private prepareFilesList(files: Array<UploadFile>): void {
+    // If the user can only select 1 image we want to reset the array
+    if ( !this.acceptMultiple) {
+      this.files = [];
+    }
+    for (const file of files) {
+      if (file.size < this.maxFileSize) {
+        if (this.acceptedTypes.includes(file.type)) {
+          this.generatePreview(file);
+          file.readableSize = this.formatBytes(file.size);
+          this.files.push(file);
+        } else {
+          const alertConfig: AlertConfig = {
+            type: AlertType.danger,
+            preMessage: 'This file type is not allowed',
+            mainMessage: `File ${file.name} is of type that is not allowed please pick one of these ${this.acceptedTypes.join(', ')}`,
+            dismissible: true,
+            timeout: this.alertService.defaultTimeout
+          };
+          this.alertService.pushAlert(alertConfig);
+          this.deleteFile(this.files.indexOf(file));
+        }
+      } else {
+        const alertConfig: AlertConfig = {
+          type: AlertType.danger,
+          preMessage: 'This file is too big',
+          mainMessage: `File ${file.name} is too big, the max size is ${this.maxFileSizeReadable}`,
+          dismissible: true,
+          timeout: this.alertService.defaultTimeout
+        };
+        this.alertService.pushAlert(alertConfig);
+        this.deleteFile(this.files.indexOf(file));
+      }
+
+    }
+  }
+
+  /**
+   * Converts the image to a base64 string
+   * @param file The file to convert
+   */
+  private generatePreview(file: UploadFile) {
+    // We can use a FileReader to generate a base64 string based on the image
+    const fileReader: FileReader = new FileReader();
+    // Convert the file to base64
```

Comment:
Sometimes you type comments this way:
`// comment`
But sometimes you do it this way:

```
/**
* Comment
*/
```
It's not really a problem, but I would suggest to use one style

(https://jsdoc.app/about-getting-started.html)

Reply:
I use // to comment in the middle of a function, it gets real messy and big if you use 3 lines for every comment. That's why I use both